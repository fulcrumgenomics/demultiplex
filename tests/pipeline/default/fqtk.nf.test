nextflow_pipeline {

    name "Test FQTK"
    script "main.nf"

    test("Should run without failures") {

        when {
            params {
                input = "$baseDir/assets/inputs/fqtk-samplesheet.csv"
                demultiplexer = 'fqtk'
                outdir = "$outputDir"
                fastq_files = "out_L001_I1_001.fastq.gz out_L001_I2_001.fastq.gz out_L001_R1_001.fastq.gz out_L001_R2_001.fastq.gz"
                read_structures = "8B 8B 150T 150T"
            }
        }

        then {
            assert workflow.success
            assert workflow.trace.succeeded().size() == 104

            assert snapshot(
                path('$outputDir/sim-data/*.fq.gz'),
                path('$outputDir/sim-data/demux-metrics.txt'),
                path('$outputDir/sim-data/unmatched*.fq.gz'),
                path("$outputDir/sim-data/Undetermined_S25_L001_1.fastp.fastq.gz")
                path("$outputDir/sim-data/Undetermined_S25_L001_2.fastp.fastq.gz")
                path("$outputDir/sim-data/s10_S10_L001_1.fastp.fastq.gz")
                path("$outputDir/sim-data/s10_S10_L001_2.fastp.fastq.gz")
                path("$outputDir/sim-data/s11_S11_L001_1.fastp.fastq.gz")
                path("$outputDir/sim-data/s11_S11_L001_2.fastp.fastq.gz")
                path("$outputDir/sim-data/s12_S12_L001_1.fastp.fastq.gz")
                path("$outputDir/sim-data/s12_S12_L001_2.fastp.fastq.gz")
                path("$outputDir/sim-data/s13_S13_L001_1.fastp.fastq.gz")
                path("$outputDir/sim-data/s13_S13_L001_2.fastp.fastq.gz")
                path("$outputDir/sim-data/s14_S14_L001_1.fastp.fastq.gz")
                path("$outputDir/sim-data/s14_S14_L001_2.fastp.fastq.gz")
                path("$outputDir/sim-data/s15_S15_L001_1.fastp.fastq.gz")
                path("$outputDir/sim-data/s15_S15_L001_2.fastp.fastq.gz")
                path("$outputDir/sim-data/s16_S16_L001_1.fastp.fastq.gz")
                path("$outputDir/sim-data/s16_S16_L001_2.fastp.fastq.gz")
                path("$outputDir/sim-data/s17_S17_L001_1.fastp.fastq.gz")
                path("$outputDir/sim-data/s17_S17_L001_2.fastp.fastq.gz")
                path("$outputDir/sim-data/s18_S18_L001_1.fastp.fastq.gz")
                path("$outputDir/sim-data/s18_S18_L001_2.fastp.fastq.gz")
                path("$outputDir/sim-data/s19_S19_L001_1.fastp.fastq.gz")
                path("$outputDir/sim-data/s19_S19_L001_2.fastp.fastq.gz")
                path("$outputDir/sim-data/s1_S1_L001_1.fastp.fastq.gz")
                path("$outputDir/sim-data/s1_S1_L001_2.fastp.fastq.gz").   
                path("$outputDir/sim-data/s20_S20_L001_1.fastp.fastq.gz")
                path("$outputDir/sim-data/s20_S20_L001_2.fastp.fastq.gz")
                path("$outputDir/sim-data/s21_S21_L001_1.fastp.fastq.gz")
                path("$outputDir/sim-data/s21_S21_L001_2.fastp.fastq.gz")
                path("$outputDir/sim-data/s22_S22_L001_1.fastp.fastq.gz")
                path("$outputDir/sim-data/s22_S22_L001_2.fastp.fastq.gz")
                path("$outputDir/sim-data/s23_S23_L001_1.fastp.fastq.gz")
                path("$outputDir/sim-data/s23_S23_L001_2.fastp.fastq.gz")
                path("$outputDir/sim-data/s24_S24_L001_1.fastp.fastq.gz")
                path("$outputDir/sim-data/s24_S24_L001_2.fastp.fastq.gz")
                path("$outputDir/sim-data/s2_S2_L001_1.fastp.fastq.gz")
                path("$outputDir/sim-data/s2_S2_L001_2.fastp.fastq.gz")
                path("$outputDir/sim-data/s3_S3_L001_1.fastp.fastq.gz")
                path("$outputDir/sim-data/s3_S3_L001_2.fastp.fastq.gz")
                path("$outputDir/sim-data/s4_S4_L001_1.fastp.fastq.gz")
                path("$outputDir/sim-data/s4_S4_L001_2.fastp.fastq.gz")
                path("$outputDir/sim-data/s5_S5_L001_1.fastp.fastq.gz")
                path("$outputDir/sim-data/s5_S5_L001_2.fastp.fastq.gz")
                path("$outputDir/sim-data/s6_S6_L001_1.fastp.fastq.gz")
                path("$outputDir/sim-data/s6_S6_L001_2.fastp.fastq.gz")
                path("$outputDir/sim-data/s7_S7_L001_1.fastp.fastq.gz")
                path("$outputDir/sim-data/s7_S7_L001_2.fastp.fastq.gz")
                path("$outputDir/sim-data/s8_S8_L001_1.fastp.fastq.gz")
                path("$outputDir/sim-data/s8_S8_L001_2.fastp.fastq.gz")
                path("$outputDir/sim-data/s9_S9_L001_1.fastp.fastq.gz")
                path("$outputDir/sim-data/s9_S9_L001_2.fastp.fastq.gz")
            ).match()
    }
}
