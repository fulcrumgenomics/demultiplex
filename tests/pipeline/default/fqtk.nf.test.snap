{
    "Should run without failures": {
        "content": [
        demux-metrics.txt:md5,1cf531103347,
        s1.R1.fq.gz:md5,c924aa78915504204e2d728bf47fbaa1,
        s1.R2.fq.gz:md5,aa81f22135ba8c4c834a3677fefec1ef,
        s10.R1.fq.gz:md5,da96b29bcbc0f1ab46679605d8b2aee7,
        s10.R2.fq.gz:md5,46af133188b882ed987878c5e6a9127d,
        s11.R1.fq.gz:md5,3bd7d2290df40b288d6f747f9614e221,
        s11.R2.fq.gz:md5,5204d1ab33424277fd03ee8015bc16b2,
        s12.R1.fq.gz:md5,92acb74d1cde5da829106495e4666108,
        s12.R2.fq.gz:md5,7639c55ac2dd001f9a0b2877ed6bff30,
        s13.R1.fq.gz:md5,d423a075715518207e0816cec386ff71,
        s13.R2.fq.gz:md5,d3d8c0de8a7c2742e7fa078f456bf12d,
        s14.R1.fq.gz:md5,27d3b688fe6fb86f7efba59811904c1a,
        s14.R2.fq.gz:md5,946fae802060b49d3112d20827ccd499,
        s15.R1.fq.gz:md5,eeb500df1cf7b5b52668a5ad76e7a4e0,
        s15.R2.fq.gz:md5,0189174a26a28312748209aa96fff994,
        s16.R1.fq.gz:md5,1d3272338381c861c6550d23b058ac0a,
        s16.R2.fq.gz:md5,de63fa4e7d41642c929d3140e4df8f20,
        s17.R1.fq.gz:md5,124bc44bd39f1cf195bbc973eaf93dcd,
        s17.R2.fq.gz:md5,824efa926b799b316fe18e3ddb96aa56,
        s18.R1.fq.gz:md5,1aeaddc40ee7d977b0f3992b1dab8df3,
        s18.R2.fq.gz:md5,7fe39e33c54098a25e99099462dc546b,
        s19.R1.fq.gz:md5,3e756430eae33bce5576cc9036db7f02,
        s19.R2.fq.gz:md5,22495830f3169e46c5140ccaea85632c,
        s2.R1.fq.gz:md5,d96a510dd672c44269907c10ccf4d5ff,
        s2.R2.fq.gz:md5,edc262ce88e69a435f6b84b608d2ec5d,
        s20.R1.fq.gz:md5,3e0274c2548b794fd8ed95ae9d892a76,
        s20.R2.fq.gz:md5,f0d0df17e27f4590e29419f385fc72be,
        s21.R1.fq.gz:md5,136e4ece90c58024de63b3717a5d5e52,
        s21.R2.fq.gz:md5,39f191679547fd28a1c88080bb38d8be,
        s22.R1.fq.gz:md5,5bc8b247c0da3123aa0a273332b4f5c9,
        s22.R2.fq.gz:md5,00fb10bbb311a8a976520854582192d1,
        s23.R1.fq.gz:md5,6bbdba0727970f512898f67afc511d70,
        s23.R2.fq.gz:md5,6cbf318479b4bf4272c272b8727d73b7,
        s24.R1.fq.gz:md5,65b778d319e0d954395b085dcd0e17cf,
        s24.R2.fq.gz:md5,1d0085033cc17ea833c30970d7c3b09f,
        s3.R1.fq.gz:md5,7f1c4996e73d083867bd2763ea4a2ee0,
        s3.R2.fq.gz:md5,bdb6d4c03207436d5184fd8117e21348,
        s4.R1.fq.gz:md5,fe9f3d22abe5155dd1a650e47c91344b,
        s4.R2.fq.gz:md5,dbc979ad76b36685a8373395f434a19f,
        s5.R1.fq.gz:md5,95c381b3bea06dd5af2d0ef3ab6eb9ff,
        s5.R2.fq.gz:md5,c360c632bebaa24f9d9a5b9a7433314e,
        s6.R1.fq.gz:md5,87ccde15e6f5915cbaf507d3be5b8203,
        s6.R2.fq.gz:md5,05b082a65be63fda0edd289f1ccbd227,
        s7.R1.fq.gz:md5,7eea579ba1346d758c878d53ee2faea6,
        s7.R2.fq.gz:md5,cc67e6f80515746e22637c4d964fd28a,
        s8.R1.fq.gz:md5,b9583720a25410840bd2af8cddaa258c,
        s8.R2.fq.gz:md5,afcb5d9df7bcbc1c5a7a2125ca4bcf0a,
        s9.R1.fq.gz:md5,650e8604780eb431a2ea174c7d702206,
        s9.R2.fq.gz:md5,b380c653c0cc390dcbab7bff9ddfa94b,
        unmatched.R1.fq.gz:md5,124410ed1b50fd9af115bf35ab31dcfc,
        unmatched.R2.fq.gz:md5,b27150c08e5cee1f4b76b4c07c136373
    ],
        "timestamp": "2023-01-18T23:52:50+0000"
    }
}